drivers/scheduler/k8s/specs/elasticsearch-aks/px-elasticdata-app.yaml

```diff
@@ -0,0 +1,152 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: elasticsearch-cluster
+spec:
+  clusterIP: None
+  selector:
+    app: es-cluster
+  ports:
+  - name: transport
+    port: 9300
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: elasticsearch-api
+spec:
+  selector:
+    app: es-cluster
+  ports:
+  - name: http
+    port: 9200
+    targetPort: 9200
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: es-config
+data:
+  elasticsearch.yml: |
+    cluster.name: my-elastic-cluster
+    network.host: "0.0.0.0"
+    bootstrap.memory_lock: false
+    discovery.zen.ping.unicast.hosts: elasticsearch-cluster
+    discovery.zen.minimum_master_nodes: 1
+    xpack.security.enabled: false
+    xpack.monitoring.enabled: false
+  ES_JAVA_OPTS: -Xms512m -Xmx512m
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: esnode
+spec:
+  serviceName: elasticsearch-api
+  replicas: 3
+  selector:
+    matchLabels:
+      app: es-cluster
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: es-cluster
+    spec:
+      schedulerName: stork
+      securityContext:
+        fsGroup: 1000
+      initContainers:
+      - name: init-sysctl
+        image: busybox
+        imagePullPolicy: IfNotPresent
+        securityContext:
+          privileged: true
+        command: ["sysctl", "-w", "vm.max_map_count=262144"]
+      containers:
+      - name: elasticsearch
+        resources:
+          requests:
+            memory: 1Gi
+        securityContext:
+          privileged: true
+          runAsUser: 1000
+          capabilities:
+            add:
+              - IPC_LOCK
+              - SYS_RESOURCE
+        image: docker.elastic.co/elasticsearch/elasticsearch:6.5.0
+        env:
+        - name: ES_JAVA_OPTS
+          valueFrom:
+            configMapKeyRef:
+              name: es-config
+              key: ES_JAVA_OPTS
+        readinessProbe:
+          httpGet:
+            scheme: HTTP
+            path: /_cluster/health?local=true
+            port: 9200
+          initialDelaySeconds: 5
+        ports:
+        - containerPort: 9200
+          name: es-http
+        - containerPort: 9300
+          name: es-transport
+        volumeMounts:
+        - name: es-data
+          mountPath: /usr/share/elasticsearch/data
+        - name: elasticsearch-config
+          mountPath: /usr/share/elasticsearch/config/elasticsearch.yml
+          subPath: elasticsearch.yml
+      volumes:
+      - name: elasticsearch-config
+        configMap:
+          name: es-config
+          items:
+          - key: elasticsearch.yml
+            path: elasticsearch.yml
+  volumeClaimTemplates:
+  - metadata:
+      name: es-data
+      annotations:
+        volume.beta.kubernetes.io/storage-class: azure-es-sc
```

Comment:
@stgleb I think storage-class is the only difference between existing specs and new spec for AKS. Can we templatize this so that we do not need to maintain 3 different app specs (AKS, AWS and for Google).

Reply:
Yes, but let it be subject of separate PR. I don't want this PR to become too big.

Reply:
Sure